From ff616b3265f5a77ed89b09a09a5c2b628e928f5c Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 30 Apr 2023 14:54:18 +0300
Subject: [PATCH] [SW-0] ppboot: add new line after critical section

during the time critical section all prints are stored in a memory
buffer and flushed out to the serial port when the time critical
section ends.
if the memory buffer is not large enough for all the prints during the
time critical section, some the last line in buffer might be cut in
the middle and the next printed line will merge with it. to avoid this
an escape sequence is printed after the buffer flush.

Change-Id: I70929a9d40646dfa9d4fd0e0e52bd8d7398880c9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 406f4ab..9d6504d 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -302,6 +302,7 @@
 {
 	preboot_uart_rbuf_flush();
 	preboot_uart_rbuf_disable();
+	preboot_puts("\n");
 }
 
 __weak void ppboot_security_init(void) {}
